src/dotnet/commands/dotnet-test/VSTestArgumentConverter.cs

```diff
@@ -0,0 +1,172 @@
+    using System.Linq;
+
+    /// <summary>
+    /// Converts the given arguments to vstest parsable arguments
+    /// </summary>
+    public class VSTestArgumentConverter
+    {
+        private const string verbosityString = "--logger:console;verbosity=";
+
+        private readonly Dictionary<string, string> ArgumentMapping = new Dictionary<string, string>
+        {
+            ["-h"] = "--help",
+            ["-s"] = "--settings",
+            ["-t"] = "--listtests",
+            ["-a"] = "--testadapterpath",
+            ["-l"] = "--logger",
+            ["-f"] = "--framework",
+            ["-d"] = "--diag",
+            ["--filter"] = "--testcasefilter",
+            ["--list-tests"] = "--listtests",
+            ["--test-adapter-path"] = "--testadapterpath",
+            ["--results-directory"] = "--resultsdirectory"
+        };
+
+        private readonly Dictionary<string, string> VerbosityMapping = new Dictionary<string, string>
+        {
+            ["q"] = "quiet",
+            ["m"] = "minimal",
+            ["n"] = "normal",
+            ["d"] = "detailed",
+            ["diag"] = "diagnostic"
+        };
+
+        private readonly string[] IgnoredArgumentArray = new string[]
+        {
+            "-c",
+            "--configuration",
+            "--runtime",
+            "-o",
+            "--output",
+            "--no-build",
+            "--no-restore",
+            "--interactive"
+        };
+
+        /// <summary>
+        /// Converts the given arguments to vstest parsable arguments
+        /// </summary>
+        /// <param name="args">original arguments</param>
+        /// <param name="ignoredArgs">arguments ignored by the converter</param>
+        /// <returns>list of args which can be passsed to vstest</returns>
+        public List<string> Convert(string[] args, out List<string> ignoredArgs)
+        {
+            var newArgList = new List<string>();
+            ignoredArgs = new List<string>();
+
+            string activeArgument = null;
+            foreach (var arg in args)
+            {
+                if (arg.StartsWith("-"))
+                {
+                    if (!string.IsNullOrEmpty(activeArgument))
+                    {
+                        if (IgnoredArgumentArray.Contains(activeArgument))
+                        {
+                            ignoredArgs.Add(activeArgument);
+                        }
+                        else
+                        {
+                            newArgList.Add(activeArgument);
+                        }
+                        activeArgument = null;
+                    }
+
+                    // Check if the arg contains the value separated by colon
+                    if (arg.Contains(":"))
+                    {
+                        var argValues = arg.Split(':');
+
+                        // Check if the argument is shortname
+                        if (ArgumentMapping.TryGetValue(argValues[0].ToLower(), out var longName))
```

Comment:
to clean up if else train, we can push Ignore Check first, followed by Verbosity